src/sourmash/sourmash_args.py

```diff
@@ -535,3 +539,177 @@ def start_file(self, filename, loader):
+        self.open()
+        return self
+
+    def __exit__(self, type, value, traceback):
+        "provide context manager functionality"
+        self.close()
+
+    def add(self, ss):
+        self.count += 1
+
+
+class SaveSignatures_NoOutput(_BaseSaveSignaturesToLocation):
+    "Do not save signatures."
+    def __repr__(self):
+        return 'SaveSignatures_NoOutput()'
+
+    def open(self):
+        pass
+
+    def close(self):
+        pass
+
+
+class SaveSignatures_Directory(_BaseSaveSignaturesToLocation):
+    "Save signatures within a directory, using md5sum names."
+    def __init__(self, location):
+        super().__init__(location)
+        
+    def __repr__(self):
+        return f"SaveSignatures_Directory('{self.location}')"
+
+    def close(self):
+        pass
+
+    def open(self):
+        try:
+            os.mkdir(self.location)
+        except FileExistsError:
+            pass
+        except:
+            notify("ERROR: cannot create signature output directory '{}'",
+                   self.location)
+            sys.exit(-1)
+
+    def add(self, ss):
+        super().add(ss)
+        md5 = ss.md5sum()
+        outname = os.path.join(self.location, f"{md5}.sig.gz")
+        with gzip.open(outname, "wb") as fp:
+            sig.save_signatures([ss], fp, compression=1)
+
+
+class SaveSignatures_SigFile(_BaseSaveSignaturesToLocation):
+    "Save signatures within a directory, using md5sum names."
+    def __init__(self, location):
+        super().__init__(location)
+        self.keep = []
+        self.compress = 0
+        if self.location.endswith('.gz'):
+            self.compress = 1
+
+    def __repr__(self):
+        return f"SaveSignatures_SigFile('{self.location}')"
+
+    def open(self):
+        pass
+
+    def close(self):
+        if self.location == '-':
+            sourmash.save_signatures(self.keep, sys.stdout)
+        else:
+            # text mode? encode in utf-8
+            mode = "w"
+            encoding = 'utf-8'
+
+            # compressed? bytes & binary.
+            if self.compress:
+                encoding = None
+                mode = "wb"
+
+            with open(self.location, mode, encoding=encoding) as fp:
+                sourmash.save_signatures(self.keep, fp,
+                                         compression=self.compress)
+
+    def add(self, ss):
+        super().add(ss)
+        self.keep.append(ss)
+
+
+class SaveSignatures_ZipFile(_BaseSaveSignaturesToLocation):
+    "Save compressed signatures in an uncompressed Zip file."
+    def __init__(self, location):
+        super().__init__(location)
+        self.zf = None
+        
+    def __repr__(self):
+        return f"SaveSignatures_ZipFile('{self.location}')"
+
+    def close(self):
+        self.zf.close()
+
+    def open(self):
+        self.zf = zipfile.ZipFile(self.location, 'w', zipfile.ZIP_STORED)
+
+    def add(self, ss):
+        assert self.zf
+        super().add(ss)
+
+        md5 = ss.md5sum()
```

Comment:
I'm worried about using `md5sum`s for zipfiles, given that we have duplicate md5sums reasonably often in our databases. 

Can we implement something to avoid overwriting duplicates?

Reply:
k, will work on it as a PR into this 'un!